includes/gateways/paypal-standard.php

```diff
@@ -334,11 +334,23 @@ function edd_process_paypal_purchase( $purchase_data ) {
  */
 function edd_listen_for_paypal_ipn() {
 	// Regular PayPal IPN
-	if ( isset( $_GET['edd-listener'] ) && $_GET['edd-listener'] == 'IPN' ) {
+	if ( isset( $_GET['edd-listener'] ) && 'ipn' === strtolower( $_GET['edd-listener'] ) ) {
 
 		edd_debug_log( 'PayPal IPN endpoint loaded' );
 
 		do_action( 'edd_verify_paypal_ipn' );
+
+		/**
+		 * This is necessary to delay execution of PayPal PDT and to avoid a race condition causing the order status
+		 * updates to be triggered twice.
+		 *
+		 * @since 2.9.4
+		 * @see https://github.com/easydigitaldownloads/easy-digital-downloads/issues/6605
+		 */
+		$token = edd_get_option( 'paypal_identity_token' );
+		if ( $token ) {
+			sleep( 5 );
```

Comment:
This needs to come before `do_action( 'edd_verify_paypal_ipn' );`

Reply:
Whoops, fixed. :)